pkg/cluster/deploybaseresources_additional.go

```diff
@@ -92,6 +93,81 @@ func (m *manager) clusterServicePrincipalRBAC() *arm.Resource {
 	)
 }
 
+func (m *manager) ensureWorkloadIdentityRBAC() ([]*arm.Resource, error) {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		return nil, nil
+	}
+
+	clusterMSI, err := m.doc.OpenShiftCluster.ClusterMsiUserAssignedIdentity()
+	if err != nil {
+		return nil, err
+	}
+
+	subnetResource, err := azure.ParseResourceID(m.doc.OpenShiftCluster.Properties.MasterProfile.SubnetID)
+	if err != nil {
+		return nil, err
+	}
+
+	resources := []*arm.Resource{}
+	vnetRG := subnetResource.ResourceGroup
+	managedRG := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+
+	resources = append(resources, m.workloadIdentityResourceGroupRBAC(rbac.RoleAzureRedHatOpenShiftFederatedCredentialRole, clusterMSI.PrincipalID, managedRG))
+	resources = append(resources, m.workloadIdentityResourceGroupRBAC(rbac.RoleAzureRedHatOpenShiftFederatedCredentialRole, clusterMSI.PrincipalID, vnetRG))
+
+	r, err := m.platoformWorkloadIdentityRBAC(vnetRG, managedRG)
+	if err != nil {
+		return nil, err
+	}
+
+	resources = append(resources, r...)
+	return resources, nil
+}
+
+func (m *manager) platoformWorkloadIdentityRBAC(vnetRG, managedRG string) ([]*arm.Resource, error) {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		return nil, nil
+	}
+
+	resources := []*arm.Resource{}
+	platformWIRolesByRoleName := m.platformWorkloadIdentityRolesByVersion.GetPlatformWorkloadIdentityRolesByRoleName()
+	platformWorkloadIdentities := m.doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile.PlatformWorkloadIdentities
+
+	for _, identity := range platformWorkloadIdentities {
+		role, exists := platformWIRolesByRoleName[identity.OperatorName]
+		if !exists {
```

Comment:
Similar to federated creation, only need to perform RBAC for identities those exists in `platformWIRolesByRoleName`.

Reply:
Ah, so if I understand correctly, we don't throw an error instead ` if !exists { continue }` ?

Reply:
Yes exactly, platformWIRolesByRoleName is just to make sure we federate and RBAC only for the desired platformIdentities.

Reply:
updated

Reply:
What about the case where there should be an entry in `platformWIRolesByRoleName` but we don't find it? Maybe we should error out in that case.

Reply:
@rajdeepc2792

Reply:
As per the discussion, this will be addressed in the new PR under a new JIRA once we come to final conclusion.